docs/troubleshoot/app-framework/zss-error-codes.md

```diff
@@ -0,0 +1,371 @@
+
+### ZWES1037E
+
+  This is usually because the server port '%d' is occupied. Is ZSS running twice?
+
+  **Reason:**
+
+  ZSS port number is already occupped. 
+
+  **Action:**
+
+  Check if another ZSS instance is already running, or chose another free port number.
+
+### ZWES1065E
+
+  Failed to configure https server, check agent https settings
+
+  **Reason:**
+
+  Failed to configure https server. 
+
+  **Action:**
+
+  Check agent https settings.
+
+
+## ZSS warning messages
+
+### ZWES1000W
+
+  Privileged server name not provided, falling back to default.
+
+  **Reason:**
+
+  Privileged server name not configured in `zowe.yaml`.
+
+  **Action:**
+
+  If your privileged server name is `ZWESIS_STD` then no action required. Otherwise set `components.zss.crossMemoryServerName` property in `zowe.yaml` to the correct name.
+
+### ZWES1005W
+
+  Plugin ID was not found in '%s'
+
+  **Reason:**
+
+  `pluginId` property wasn't found in `<path-to-pluginDefinition.json>` file.
+
+  **Action:**
+
+  Add `pluginId` property into `<path-to-pluginDefinition.json>` file.
+
+### ZWES1012W
+
+  Could not open pluginsDir '%s': Ret='%d', res='0x%x'
+
+  **Reason:**
+
+  Could not open `<pluginsDir>`: Ret=`<return-code>`, res=`<reason-code>`
+
+  **Action:**
+
+  Check that `<pluginsDir>` exists and allows reading.
+
+### ZWES1060W
+
+  Failed to init TLS environment, rc=%d(%s)
+
+  **Reason:**
+
+  Failed to initialized TLS environment.
+
+  **Action:**
+
+  Ensure that ZSS certificate is configured correctly.
+
+### ZWES1103W
+
+  Product Registration failed, RC = %d
+
+  **Reason:**
+
+  Failed to register ZSS.
+
+  **Action:**
+
+  No action required.
+
+### ZWES1201W
+
+  Could not %s file '%s': Ret='%d', res='%d'
+
+  **Reason:**
+
+  Unixfile REST Service could not `<action>` file `<filename>`: Ret=`<return-code>`, res=`<reason-code>`
+
+  **Action:**
+
+  No action required.
+
+### ZWES1103W
+
+  Could not get metadata for file '%s': Ret='%d', res='%d'
+
+  **Reason:**
+
+  Unixfile REST Service could not get metadata for file `<filename>`: Ret=`<return-code>`, res=`<reason-code>`
+
+  **Action:**
+
+  No action required.
+
+### ZWES1400W
+
+  Non standard class provided for '%s' '%s', ending request...
+
+  **Reason:**
+
+  Non standard class provided in the request for Security service. The request was cancelled.
+
+  **Action:**
+
+  No action required.
+
+
+### ZWES1602W
+
+  JWK is in unrecognized format
+
+  **Reason:**
+
+  JSON Web Key(JWK) is in unrecognized format. 
+
+  **Action:**
+
+  No action required.
+
+### ZWES1603W
+
+  Failed to construct public key using JWK
+
+  **Reason:**
+
+  JSON Web Key(JWK) has invalid public key info. 
+
+  **Action:**
+
+  No action required.
+
+### ZWES1604W
+
+  JWK: failed to init HTTP context, ensure that APIML and TLS settings are correct
+
+  **Reason:**
+
+  Failed to init HTTP context for requesting JSON Web Key(JWK).
+
+  **Action:**
+
+  Check that your certificate setup.
```

Comment:
Check the zowe keystore configuration and specification of it within the zowe server config.